pkg/yaml/yaml_test.go

```diff
@@ -232,22 +186,15 @@ func TestWriteYamlFile(t *testing.T) {
 	// Setup temporary file
 	file, err := CreateTempFile(t, yamlFileNamePattern)
 	defer file.Close()
-	if err != nil {
-		t.Errorf("Can't create file: %s", file.Name())
-	}
+	assert.NoError(t, err, "Can't create file: %s", file.Name())
 
 	//Check if writing a single YAML works
 	writeYamlFileErr := WriteYamlFile(file.Name(), yaml)
-	if writeYamlFileErr != nil {
-		t.Errorf("Error while trying to write YAML to file")
-	}
+	assert.NoError(t, writeYamlFileErr, "Error while trying to write YAML to file")
+
 	b, err := os.ReadFile(file.Name())
-	if err != nil {
-		t.Errorf("Error while reading file for evaluation")
-	}
-	if string(b) != expectedString {
-		t.Errorf("Yaml not written correctly. Should be \n%s\nbut is:\n%s\n", expectedString, string(b))
-	}
+	assert.NoError(t, err, "Error while reading file for evaluation")
+	assert.Equal(t, expectedString, string(b), "Yaml not written correctly. Should be \n%s\nbut is:\n%s\n", expectedString, string(b))
```

Comment:
no need to format the error string with details. assert.Equal will show a nicely readable diff of the expected vs actual value

Reply:
Already fixed in https://github.com/kluctl/kluctl/pull/241/commits/94cd0459542be6c733eefde43be027061ee41e15